tests/components/zha/test_switch.py

```diff
@@ -507,3 +562,125 @@ async def test_switch_configurable(
+WCCS = closures.WindowCovering.ConfigStatus
+WCM = closures.WindowCovering.WindowCoveringMode
+
+
+async def test_cover_inversion_switch(
+    hass: HomeAssistant, zha_device_joined_restored, zigpy_cover_device
+) -> None:
+    """Test ZHA cover platform."""
+
+    # load up cover domain
+    cluster = zigpy_cover_device.endpoints.get(1).window_covering
+    cluster.PLUGGED_ATTR_READS = {
+        WCAttrs.current_position_lift_percentage.name: 65,
+        WCAttrs.current_position_tilt_percentage.name: 42,
+        WCAttrs.window_covering_type.name: WCT.Tilt_blind_tilt_and_lift,
+        WCAttrs.config_status.name: WCCS(~WCCS.Open_up_commands_reversed),
+        WCAttrs.window_covering_mode.name: WCM(WCM.LEDs_display_feedback),
+    }
+    update_attribute_cache(cluster)
+    zha_device = await zha_device_joined_restored(zigpy_cover_device)
+    assert (
+        not zha_device.endpoints[1]
+        .all_cluster_handlers[f"1:0x{cluster.cluster_id:04x}"]
+        .inverted
+    )
+    assert cluster.read_attributes.call_count == 3
+    assert (
+        WCAttrs.current_position_lift_percentage.name
+        in cluster.read_attributes.call_args[0][0]
+    )
+    assert (
+        WCAttrs.current_position_tilt_percentage.name
+        in cluster.read_attributes.call_args[0][0]
+    )
+
+    entity_id = find_entity_id(Platform.SWITCH, zha_device, hass)
+    assert entity_id is not None
+
+    await async_enable_traffic(hass, [zha_device], enabled=False)
+    # test that the cover was created and that it is unavailable
+    state = hass.states.get(entity_id)
+    assert state
+    assert state.state == STATE_UNAVAILABLE
+
+    # allow traffic to flow through the gateway and device
+    await async_enable_traffic(hass, [zha_device])
+    await hass.async_block_till_done()
+
+    # test update
+    prev_call_count = cluster.read_attributes.call_count
+    await async_update_entity(hass, entity_id)
+    assert cluster.read_attributes.call_count == prev_call_count + 1
+    state = hass.states.get(entity_id)
+    assert state
+    assert state.state == STATE_OFF
+
+    # test to see the state remains after tilting to 0%
+    await send_attributes_report(
+        hass, cluster, {WCAttrs.current_position_tilt_percentage.id: 0}
+    )
+    state = hass.states.get(entity_id)
+    assert state
+    assert state.state == STATE_OFF
+
+    with patch(
+        "zigpy.zcl.Cluster.write_attributes", return_value=[0x1, zcl_f.Status.SUCCESS]
+    ):
+        cluster.PLUGGED_ATTR_READS = {
+            WCAttrs.config_status.name: WCCS.Operational
+            | WCCS.Open_up_commands_reversed,
+        }
+        # turn on from UI
+        await hass.services.async_call(
+            SWITCH_DOMAIN, "turn_on", {"entity_id": entity_id}, blocking=True
+        )
+        assert cluster.write_attributes.call_count == 1
+        assert cluster.write_attributes.call_args_list[0] == call(
+            {
+                "window_covering_mode": WCM.Motor_direction_reversed
```

Comment:
minor too, but we could use `WCAttrs.window_covering_mode.name` here.

(In general, attribute defs should probably be cleaned up in another PR for all tests.)

Reply:
cleaned up ones from this PR